monotonic.py

```diff
@@ -0,0 +1,330 @@
+#------------------------------#
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import csv
+from pathlib import Path
+
+import monotonic_functions
+#------------------------------#
+                                                                                
+def monotonic(input_dir, output_dir, legendvar,YS02var):
+    """Takes in two directories: input location (where the data is) and output 
+    location (where to save results) and performs data analysis for monotonic 
+    test data. Returns material parameters and plots"""
+
+    print("************************************")
+    print("Running Monotonic Analysis")
+
+    #Define in and out Directories, make new folder for output ...
+    input_directory = input_dir
+    output_directory = Path(output_dir,"Monotonic_Results")
+    os.mkdir(output_directory)
+
+    #Make the output directory name a string for use later on ...
+    output_directory = str(output_directory) 
+    
+    #Make folder for the plots to be saved to ...
+    plots_directory = Path(output_directory,"Plots")
+    os.mkdir(plots_directory)
+    plots_directory = str(plots_directory)
+    
+    #"Unpack" data files from input directory ...
+    print("Reading Input Data Files...")
+    mfiles = next(os.walk(input_directory))[2]
+    nfiles = len(mfiles) #Number of files in directory
+        
+    #Initiate all output variables for iteration
+    testnames = [] #Names of test files
+    E = [] #Elastic Modulus
+    Pos1 = [] #Starting Position
+    Pos2 = [] #End Position
+    Nom_Ext = [] #Nominal Extension (end-start)
+    MaxLoad = [] #Maximum recorded load
+    MaxStrain = [] #Maximum recorded axial strain
+    Max_Trans = [] #Maximum recorded transverse strain
+    v = [] #Poisson's Ratio
+    True_Strength = [] #Ultimate true stress
+    True_Ductility = [] #Maximum true strain
+    Engr_FracStrength = [] #Engineering stress @ failure
+    YS02 = [] #0.2% offset strength
+    YE02 = [] #0.2% offset strain
+    YE02_P = [] #0.2% offset % elong.
+    YS =[] #Yield Strength
+    YE = [] #Yield Strain
+    RA = [] # %-Reduction-Area
+    
+    #These will be used later to collect all data for 
+    # overlaid stress vs strain plot ...
+    totalstress = []
+    totalstrain = []
+    
+    #Main Monotonic Analysis Loop...
+    for j in range(nfiles):
+        #Add each file name to the end of the directory location
+        location = Path(input_directory, mfiles[j])
+       
+        filename = str(mfiles[j])
+        filename = filename[0:-4]
+        print("    Reading File ",filename)
+
+        #Read in the Data set from every test individually...
+        mData = pd.read_csv(location,header=None) 

```

Comment:
To make things more universal you may want there to be headers. 

- If the data has headers you can define them 
- Or you could create a list of headers that are in the correct order and add them in there